tests/test_biascorr.py

```diff
@@ -0,0 +1,518 @@
+
+        # Try default "fit" parameters instantiation
+        dirbias = biascorr.DirectionalBias(angle=45)
+
+        assert dirbias._fit_or_bin == "bin_and_fit"
+        assert dirbias._meta["fit_func"] == biascorr.fit_workflows["nfreq_sumsin"]["func"]
+        assert dirbias._meta["fit_optimizer"] == biascorr.fit_workflows["nfreq_sumsin"]["optimizer"]
+        assert dirbias._meta["angle"] == 45
+
+    @pytest.mark.parametrize("angle", [20, 90, 210])  # type: ignore
+    @pytest.mark.parametrize("nb_freq", [1, 2, 3])  # type: ignore
+    def test_directionalbias__synthetic(self, angle, nb_freq) -> None:
+        """Test the subclass DirectionalBias with synthetic data."""
+
+        # Get along track
+        xx = gu.raster.get_xy_rotated(self.ref, along_track_angle=angle)[0]
+
+        # Get random parameters (3 parameters needed per frequency)
+        np.random.seed(42)
+        params = np.array([(5, 3000, np.pi), (1, 300, 0), (0.5, 100, np.pi / 2)]).flatten()
+        nb_freq = 1
+        params = params[0 : 3 * nb_freq]
+
+        # Create a synthetic bias and add to the DEM
+        synthetic_bias = sumsin_1d(xx.flatten(), *params)
+        bias_dem = self.ref - synthetic_bias.reshape(np.shape(self.ref.data))
+
+        # For debugging
+        if PLOT:
+            synth = self.ref.copy(new_array=synthetic_bias.reshape(np.shape(self.ref.data)))
+            import matplotlib.pyplot as plt
+
+            synth.show()
+            plt.show()
+
+            dirbias = biascorr.DirectionalBias(angle=angle, fit_or_bin="bin", bin_sizes=10000)
+            dirbias.fit(reference_dem=self.ref, dem_to_be_aligned=bias_dem, subsample=10000, random_state=42)
+            xdem.spatialstats.plot_1d_binning(
+                df=dirbias._meta["bin_dataframe"], var_name="angle", statistic_name="nanmedian", min_count=0
+            )
+            plt.show()
+
+        # Try default "fit" parameters instantiation
+        dirbias = biascorr.DirectionalBias(angle=angle, bin_sizes=300)
+        bounds = [
+            (2, 10),
+            (500, 5000),
+            (0, 2 * np.pi),
+            (0.5, 2),
+            (100, 500),
+            (0, 2 * np.pi),
+            (0, 0.5),
+            (10, 100),
+            (0, 2 * np.pi),
+        ]
+        dirbias.fit(
+            reference_dem=self.ref,
+            dem_to_be_aligned=bias_dem,
+            subsample=10000,
+            random_state=42,
+            bounds_amp_wave_phase=bounds,
+            niter=10,
+        )
+
+        # Check all parameters are the same within 10%
+        fit_params = dirbias._meta["fit_params"]
+        assert np.shape(fit_params) == np.shape(params)
+        assert np.allclose(params, fit_params, rtol=0.1)
+
+        # Run apply and check that 99% of the variance was corrected
+        corrected_dem = dirbias.apply(bias_dem)
+        assert np.nanvar(corrected_dem - self.ref) < 0.01 * np.nanvar(synthetic_bias)
+
+    def test_deramp(self) -> None:
```

Comment:
This test doesn't really seem to do anything other than making sure it was instantiated correctly. If that's intentional, perhaps call it `test_deramp_instantiation()`

Reply:
Following the logic above, no option is passed here so the test implicitly only calls the function/class with default arguments.